3_Fixed_Sine_Frequencies

thehale opened the pull request:
Fixes eriknyquist/tones#3

Resolved the issue where some input sine frequencies were processed into
the same output frequency.

No work/testing was done to see how this change affected
other waveforms.

thehale committed:
3_Fixed_Sine_Frequencies

Fixes eriknyquist/tones#3

Resolved the issue where some input sine frequencies were processed into
the same output frequency.

No work/testing was done to see how this change affected
other waveforms.

eriknyquist-avive commented:
Awesome, nice work! I need to test this out and make sure it doesn't break anything else, but the code looks good from what I can see. I will be away for the next 4 days, so I will probably not get to this pull request until early next week.

thehale committed:
3_Fixed_Other_Frequencies

As expected, my last commit broke all the other wave types and the
note-based compositions. This commit fixes all of those issues so that
the module behaves just as it did before I touched it, but with more
accurate frequencies than before.

eriknyquist commented:
Was just testing this out, the sine waves and square waves sound great, but there is something wrong with the triangle and sawtooth waves.

I have attached two audio files, one playing a song with a sine wave (which sounds fine), and another playing the same song with a triangle wave (sounds terrible :D).

I had a look at the code in _triangle_wave_samples, and I tried changing the `period = int(rate / freq)` statement to `period = float(rate) / float(freq)`, but this did not fix the problem. Aside from that change, the code looks fine to me, so I'm not sure what's wrong yet. I will take another look tomorrow night.

[jhale_test_audio.zip](https://github.com/eriknyquist/tones/files/5611579/jhale_test_audio.zip)

eriknyquist commented:
Found the problem, I'll fix it separately. It was an existing issue with how I'm generating triangle & sawtooth waveforms.
